### PR TITLE
Incorporate ConfigProvider in the queue

### PR DESCRIPTION
**Pull request by szwedm**
**Description**

Changes proposed in this pull request:

- incorporate ConfigProvider in KEB's workflow,
- get RuntimeVersionData where necessary (ConfigProvider requirement)
- adjust tests - add ConfigProvider to the tested flows.

This is part 2 of #1875 

**Related issue(s)**
Issue: #1824

### CODE REVIEW

File: components/kyma-environment-broker/cmd/broker/broker_suite_test.go [hunk 1]
```diff
@@ -122,6 +123,11 @@ func NewBrokerSuiteTest(t *testing.T, version ...string) *BrokerSuiteTest {
 	componentsYAML := kebRuntime.ReadYAMLFromFile(t, "kyma-components.yaml")
 	fakeHTTPClient := kebRuntime.NewTestClient(t, installerYAML, componentsYAML, http.StatusOK)
 
+	configProvider := kebConfig.NewConfigProvider(
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/cmd/broker/broker_suite_test.go:126]: I'd like to see one constructor which initializes inside the reader and validator, so on the high level, you will have kebConfig.NewConfigProvider(ctx, cli, logrus.New())

File: components/kyma-environment-broker/internal/process/update/initialisation_step.go [hunk 2]
```diff
@@ -28,19 +29,29 @@ const (
...
 type InitialisationStep struct {
-	operationManager *process.UpdateOperationManager
-	operationStorage storage.Operations
-	instanceStorage  storage.Instances
-	inputBuilder     input.CreatorForPlan
+	operationManager       *process.UpdateOperationManager
+	operationStorage       storage.Operations
+	instanceStorage        storage.Instances
+	runtimeStatesStorage   storage.RuntimeStates
+	runtimeVerConfigurator RuntimeVersionConfiguratorForUpdating
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/update/initialisation_step.go:42]: runtimeStateStorage and runtimeVerConfigurator - where is it used?

File: components/kyma-environment-broker/internal/process/update/initialisation_step.go [hunk 3]
```diff
@@ -100,9 +127,18 @@ func (s *InitialisationStep) Run(operation internal.UpdatingOperation, log logru
 	return s.initializeUpgradeShootRequest(operation, log)
 }
 
+func (s *InitialisationStep) getRuntimeIdFromProvisioningOp(operation *internal.UpdatingOperation, log logrus.FieldLogger) error {
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/process/update/initialisation_step.go:130]: log not used

File: components/kyma-environment-broker/cmd/broker/broker_suite_test.go (hunk 1 above)
Comment (szwedm) [reply components/kyma-environment-broker/cmd/broker/broker_suite_test.go:126]: The idea was to explicitly show where the configuration comes from and how it's provided. On first sight we can see that ConfigProvider provides the configuration from a ConfigMap. I left some space for future implementations, for example, configuration provided from a file or some other endpoint. I think we can hide the components initialised in NewConfigProvider() if ConfigMap stays as a default source of the configuration.

File: components/kyma-environment-broker/internal/process/update/initialisation_step.go (hunk 2 above)
Comment (szwedm) [reply components/kyma-environment-broker/internal/process/update/initialisation_step.go:42]: nowhere, this is a leftover from the one of options I was trying to implement for fetching last runtimeID. Thanks for spotting it, will be removed

File: components/kyma-environment-broker/internal/process/update/initialisation_step.go (hunk 3 above)
Comment (szwedm) [reply components/kyma-environment-broker/internal/process/update/initialisation_step.go:130]: oversight from extracting method using IDE, thanks for spotting it, will be removed

File: components/kyma-environment-broker/cmd/broker/broker_suite_test.go (hunk 1 above)
Comment (szwedm) [reply components/kyma-environment-broker/cmd/broker/broker_suite_test.go:126]: We discussed the matter and decided to keep the proposed solution

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**